src/learn/learn.cpp

```diff
@@ -665,6 +668,44 @@ namespace Learner
         limits.depth = 0;
     }
 
+    PSVector LearnerThink::fetch_next_validation_set()
+    {
+        PSVector validation_data;
+
+        auto mainThread = Threads.main();
+        mainThread->execute_with_worker([&validation_data, this](auto& th){
+            auto do_include_predicate = [&th, this](const PackedSfenValue& ps) -> bool {
+                if (params.eval_limit < abs(ps.score))
+                    return false;
+
+                if (!params.use_draw_games_in_validation && ps.game_result == 0)
+                    return false;
+
+                if (params.smart_fen_skipping_for_validation)
+                {
+                    StateInfo si;
+                    auto& pos = th.rootPos;
+                    if (pos.set_from_packed_sfen(ps.sfen, &si, &th) != 0)
+                        return false;
+
+                    if (pos.capture_or_promotion((Move)ps.move) || pos.checkers())
```

Comment:
I guess that conditions will be added in the future.  It would be annoying to add theme here and https://github.com/nodchip/Stockfish/pull/274/files#diff-824cc3b1e082047f44138fee8ec9d88b5e2e383b713cea2275f6e98a4e029f44R849.  Could you please create a function to determine if a position should be skipped or not, and call here and https://github.com/nodchip/Stockfish/pull/274/files#diff-824cc3b1e082047f44138fee8ec9d88b5e2e383b713cea2275f6e98a4e029f44R849?

Reply:
The conditions may and will be slightly different for validation data than for training data. Right now I'm thinking whether it would make sense to exclude positions with a single legal move and positions which have checking moves. I'm thinking about these two conditions because such positions don't make sense for move accuracy computation.

Reply:
> The conditions may and will be slightly different for validation data than for training data.

I missed it.  Thank you for the explanation.

> Right now I'm thinking whether it would make sense to exclude positions with a single legal move and positions which have checking moves. I'm thinking about these two conditions because such positions don't make sense for move accuracy computation as we do it now.

I thought about it but I don't have any conclusions.  Umm...